### DISCUSSION TITLE
Comments & Subscribe

### DISCUSSION THREAD

**Discussion by satrox28** [thuliteio/doks#1269] [Q&A]
Hi,

I am looking for options to allow comments on the blog posts and enable subscribe to feed all the posts to be sent to end users in a regular schedule.

**Comment by satrox28** [comment]
@h-enk Can you share your inputs.

**Reply by h-enk** [reply]
I've only used [giscus](https://giscus.app/) for comments and [Keila](https://www.keila.io/) for newsletters (using [Netlify Forms](https://www.netlify.com/platform/core/forms/)). And, Mailchimp's [RSS-to-Email](https://mailchimp.com/features/rss-to-email/) functionality. All worked great for me, but there are more options out there.

Please check the [Hugo forum](https://discourse.gohugo.io/)

**Comment by satrox28** [comment]
Thanks for the info. Do any using disqus?

**Reply by h-enk** [reply]
Note that [Hugo ships with an internal Disqus template](https://gohugo.io/content-management/comments/)

**Reply by satrox28** [reply]
Thanks. In that case hope I can update config/_default/hugo.toml?.

**Reply by h-enk** [reply]
Yes, that should work!

**Reply by satrox28** [reply]
I added it on the config/_default/config.toml
```
[services]
  [services.disqus]
    shortname = '<disqusName>'
```

But I don't see any comments on the blog posts. @h-enk

**Reply by h-enk** [reply]
Did you also add the internal template to `layouts/blog/single.html` (copied from `node_modules/@hyas/doks-core/layouts/blog/single.html`)?

For example, after line 34:

```html
{{ template "_internal/disqus.html" . }}
```

**Reply by satrox28** [reply]
Yes, i tried to do it and now am getting below error.

```
ERROR render of "page" failed: execute of template failed: template: _internal/disqus.html:1:16: executing "_internal/disqus.html" at <.Site.Config.Privacy.Disqus>: can't evaluate field Site in type page.Pages
```

I tried to add the privacy field but it didn't help.

```
[privacy]
  [privacy.disqus]
    disable = false
```

Looks like some property on the config.toml is missing. Can you help @h-enk, thanks :)

**Reply by h-enk** [reply]
I can't reproduce your issue /it renders fine for me:

![2024-09-02_09-32-50](https://github.com/user-attachments/assets/7bb2ef73-1ebb-4850-ae4a-e0387b0903fb)

Note, I put this _at the bottom_ of `config/_default/hugo.toml`:

```toml

[services]
  [services.disqus]
    shortname = 'your-disqus-shortname'

[privacy]
  [privacy.disqus]
    disable = false
```
